.github/workflows/ubuntu.yml

```diff
@@ -0,0 +1,37 @@
+name: Build on ubuntu
+
+on:
+  push:
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: true
+    - run: |
+        git config --global --add safe.directory '*'
+        git submodule update --init --recursive
```

Comment:
Are we expected to run: `git submodule update --init --recursive` manually, I was hoping that `actions/checkout` would take care of submodules.

Reply:
for now this is needed

Reply:
Please take a look at the [action page](https://github.com/actions/checkout#usage)
it seems like there's a cleaner way of doing this.